TCMSApp/src/client/app/blocks/fakers/fakeRuns.service.js

```diff
@@ -0,0 +1,125 @@
+                "status": "string: [passed, executing, failed]",
+                "tests": [
+                    {
+                        "id": "int",
+                        "status": "boolean",
+                        "steps": [
+                            {
+                                "number": "int",
+                                "action": "string",
+                                "expected": "string",
+                                "status": "string: [passed, blocked, failed]"
+                            }
+                        ]
+                    }
+                ]
+            }
+         *
+         */
+        this.getFakeRuns = function (quantity) {
+            var quantity = quantity || 1;
+            var runs = [];
+            var runStatuses = ['passed', 'executing', 'failed'];
+            var testStatuses = ['passed', 'executing', 'failed'];
+            var stepStatuses = ['passed', 'blocked', 'failed'];
+
+            /**
+             * @memberOf getFakeRuns
+             * @function getSteps
+             * @param quantity
+             * @returns {Array}
+             */
+            function getSteps(quantity) {
+                var steps = [];
+                let i = -1;
+                while (++i < quantity) {
+                    let step = {
+                        number: faker.random.number(),
+                        action: faker.lorem.sentence(),
+                        expected: faker.lorem.sentence(),
+                        status: stepStatuses[faker.random.number(stepStatuses.length - 1)]
+                    };
+
+                    steps.push(step);
+                }
+                return steps;
+            }
+
+            /**
+             * @memberOf getFakeRuns
+             * @function getTests
+             * @param quantity
+             * @returns {Array}
+             */
+            function getTests(quantity) {
+                var tests = [];
+                let i = -1;
+                while (++i < quantity) {
+                    let test = {
+                        id: faker.random.number(1000),
+                        status: faker.random.boolean(),
+                        steps: getSteps(faker.random.number(10))
+                    };
+
+                    tests.push(test);
+                }
+                return tests;
+            }
+
+            /**
+             * @memberOf getFakeRuns
+             * @name i
+             * @description loop counter
+             * @type {number}
+             */
+            let i = -1;
+            while (++i < quantity) {
+                /**
+                 *
+                 * @type {{_id: *, previousRunId: *, dateStart: *, dateEnd: *, build: *, environment: string, status: string, tests: Array}}
+                 */
+                let run = {
+                    _id: faker.random.number(1000000),
+                    previousRunId: faker.random.number(1000000),
```

Comment:
I am thinking about the next thing: we might generate wrong data because previousRunId will be the link to some run with some random _id. We have no guarantee that that particular _id will be generated. 
- [ ] Can you please start generating _id's from 1 and consider that fact while generating previousRunId  
- [ ] And also not every run should have parent.